experimental/segmenter/src/lstm_bies.rs

```diff
@@ -48,40 +68,48 @@ impl<'l> Lstm<'l> {
             return Err(Error::Syntax);
         }
 
-        let mat1 = data.get().mat1.as_matrix_zero::<2>()?;
-        let mat2 = data.get().mat2.as_matrix_zero::<3>()?;
-        let mat3 = data.get().mat3.as_matrix_zero::<3>()?;
-        let mat4 = data.get().mat4.as_matrix_zero::<2>()?;
-        let mat5 = data.get().mat5.as_matrix_zero::<3>()?;
-        let mat6 = data.get().mat6.as_matrix_zero::<3>()?;
-        let mat7 = data.get().mat7.as_matrix_zero::<2>()?;
-        let mat8 = data.get().mat8.as_matrix_zero::<3>()?;
-        let mat9 = data.get().mat9.as_matrix_zero::<1>()?;
-        let embedd_dim = mat1.dim().1;
-        let hunits = mat3.dim().0;
-        if mat2.dim() != (hunits, 4, embedd_dim)
-            || mat3.dim() != (hunits, 4, hunits)
-            || mat4.dim() != (hunits, 4)
-            || mat5.dim() != (hunits, 4, embedd_dim)
-            || mat6.dim() != (hunits, 4, hunits)
-            || mat7.dim() != (hunits, 4)
-            || mat8.dim() != (2, 4, hunits)
-            || mat9.dim() != (4)
+        #[cfg(debug_assertions)]
+        for &unaligned in data.get().dic.iter_values() {
+            if i16::from_unaligned(unaligned) as usize >= data.get().dic.len() {
+                return Err(Error::DimensionMismatch);
+            }
+        }
+
+        let embedding = data.get().embedding.as_matrix_zero::<2>()?;
+        let fw_w = data.get().fw_w.as_matrix_zero::<3>()?;
+        let fw_u = data.get().fw_u.as_matrix_zero::<3>()?;
+        let fw_b = data.get().fw_b.as_matrix_zero::<2>()?;
+        let bw_w = data.get().bw_w.as_matrix_zero::<3>()?;
+        let bw_u = data.get().bw_u.as_matrix_zero::<3>()?;
+        let bw_b = data.get().bw_b.as_matrix_zero::<2>()?;
+        let time_w = data.get().time_w.as_matrix_zero::<3>()?;
+        let time_b = data.get().time_b.as_matrix_zero::<1>()?;
+        let embedd_dim = embedding.dim().1;
+        let hunits = fw_u.dim().0;
+        if embedding.dim() != (data.get().dic.len() + 1, embedd_dim)
```

Comment:
Suggestion: make `data.get().dic.len()` a new variable named `num_classes` or similar.

Reply:
`len` is a free call and the value shouldn't be duplicated.